pkg/kaniko/context_remote.go

```diff
@@ -54,29 +56,81 @@ func (c RemoteContextProvider) PrepareContext(opts types.ImageBuilderOpts) (stri
 func createArchive(buildContextDir string, tarGzPath string) error {
 	logger.Infof("Creating docker build-context for kaniko")
 
-	tarGzArchiver := archiver.TarGz{
-		Tar: &archiver.Tar{
-			OverwriteExisting:      true,
-			MkdirAll:               true,
-			ImplicitTopLevelFolder: false,
-			ContinueOnError:        false,
-		},
-		CompressionLevel: gzip.BestCompression,
+	// Check if the build context directory exists.
+	if _, err := os.Stat(buildContextDir); os.IsNotExist(err) {
+		return fmt.Errorf("can't access directory %q: it doesn't exist", buildContextDir)
 	}
 
-	var filesToArchive []string
+	// Walk through the build context directory, and collect all the files to be archived.
+	files := make(map[string]os.FileInfo)
+	if err := filepath.Walk(buildContextDir, func(filePath string, info os.FileInfo, err error) error {
+		if err != nil {
```

Comment:
which `err` are you checking here ?

Reply:
It is to prevent panic by handling failure accessing a path. An example here:
https://pkg.go.dev/path/filepath#example-Walk